src/main/java/seedu/pennywise/model/entry/Tag.java

```diff
@@ -75,27 +123,29 @@ public static boolean isValidTagName(EntryType type, String test) {
      * Gets tagName
      */
     public String getTagName() {
-        return tagName;
+        String name = tag.toString();
+        String camelCaseName = name.substring(0, 1) + name.substring(1).toLowerCase();
```

Comment:
I think instead of concatenating the substring we can define methods in enums like:
https://www.baeldung.com/a-guide-to-java-enums
```
public enum IncomeTag implements EntryTag {
        SALARY("Salary"),
        ...
        
        private final String tagName;

        IncomeTag(String tagName) {
            this.tagName = tagName;
        }

        @Override
        public String toString() {
                return tagName;
        }
}
```